pandas/core/series.py

```diff
@@ -4691,8 +4731,16 @@ def rename(
             axis = self._get_axis_number(axis)
 
         if callable(index) or is_dict_like(index):
+            # error: Argument 1 to "_rename" of "NDFrame" has incompatible type
+            # "Union[Union[Mapping[Hashable, Any], Callable[[Hashable], Hashable]],
+            # Hashable, None]"; expected "Union[Mapping[Hashable, Any],
+            # Callable[[Hashable], Hashable], None]"
```

Comment:
I think (but I'm not sure) this can be avoided by just adding the line
```python
assert not isinstance(index, Hashable)
```

Reply:
I would rather keep the ignore-comment since many objects are Hashable (Typeguards might help in the future). Even lambda functions are Hashable:
`"__hash__" in dir(lambda: x)  # True`

Reply:
OK, that makes sense.  The `lambda` would also match `Callable`, but I guess we have to live with this aspect of python typing.

Reply:
does the comment (error message) need updating?

Reply:
> does the comment (error message) need updating?

done